iib/workers/tasks/build_containerized_merge.py

```diff
@@ -0,0 +1,376 @@
+                _binary_image=binary_image,
+                overwrite_target_index_token=overwrite_target_index_token,
+                source_from_index=source_from_index,
+                target_index=target_index,
+                distribution_scope=distribution_scope,
+                binary_image_config=binary_image_config,
+            ),
+        )
+
+    source_from_index_resolved = prebuild_info['source_from_index_resolved']
+    target_index_resolved = prebuild_info['target_index_resolved']
+
+    # Set OPM version
+    Opm.set_opm_version(target_index_resolved)
+    opm_version = Opm.opm_version
+
+    _update_index_image_build_state(request_id, prebuild_info)
+
+    mr_details: Optional[Dict[str, str]] = None
+    local_git_repo_path: Optional[str] = None
+    index_git_repo: Optional[str] = None
+    last_commit_sha: Optional[str] = None
+    output_pull_spec: Optional[str] = None
+    original_index_db_digest: Optional[str] = None
+
+    with tempfile.TemporaryDirectory(prefix=f'iib-{request_id}-') as temp_dir:
+        # Setup and clone Git repository
+        branch = prebuild_info['ocp_version']
+        (
+            index_git_repo,
+            local_git_repo_path,
+            localized_git_catalog_path,
+        ) = prepare_git_repository_for_build(
+            request_id=request_id,
+            from_index=source_from_index,
+            temp_dir=temp_dir,
+            branch=branch,
+            index_to_gitlab_push_map=index_to_gitlab_push_map or {},
+        )
+
+        # Pull both source and target index.db artifacts and read present bundle
+        source_index_db_path = fetch_and_verify_index_db_artifact(
+            source_from_index_resolved, temp_dir
+        )
+        target_index_db_path = fetch_and_verify_index_db_artifact(target_index_resolved, temp_dir)
+
+        # Get the bundles from the index.db file
+        with set_registry_token(overwrite_target_index_token, target_index, append=True):
+            source_index_bundles, source_index_bundles_pull_spec = _get_present_bundles(
+                source_index_db_path, temp_dir
+            )
+            log.debug("Source index bundles %s", source_index_bundles)
+            log.debug("Source index bundles pull spec %s", source_index_bundles_pull_spec)
+
+            target_index_bundles, target_index_bundles_pull_spec = _get_present_bundles(
+                target_index_db_path, temp_dir
+            )
+            log.debug("Target index bundles %s", target_index_bundles)
+            log.debug("Target index bundles pull spec %s", target_index_bundles_pull_spec)
+
+        # Validate the bundles from source and target have their pullspecs present in the registry
+        set_request_state(
+            request_id,
+            'in_progress',
+            'Validating whether the bundles have their pullspecs present in the registry',
+        )
+        unique_bundles = list(set(source_index_bundles + target_index_bundles))
```

Comment:
should this be `target_index_bundles_pull_spec` and `source_index_bundles_pullspec` instead?

Reply:
I do not see ValidateBundlesThread accessing bundle['bundlePath'], so it'll fail if target_index_bundles is passed as it'll be a `BundleImage` dict

Reply:
@JAVGan why do extra computation in `validate_bundles_in_parallel` when you have the list of bundle pullspecs readily available in `target_index_bundles_pull_spec` and `source_index_bundles_pullspec`?